src/models/Factor.js

```diff
@@ -235,9 +240,17 @@ function (Okta, Q, factorUtil, BaseLoginModel) {
 
         return promise
         .then(function (trans) {
+          var options = {
+            'delay': PUSH_INTERVAL
+          };
           if (trans.status === 'MFA_CHALLENGE' && trans.poll) {
             return Q.delay(PUSH_INTERVAL).then(function() {
-              return trans.poll(PUSH_INTERVAL);
+              if (self.settings.get('features.autoPush') && self.get('factorType') === 'push') {
+                options.autoPush = _.bind(function() {
+                  return self.get('autoPush');
```

Comment:
shouldn't this be ```return this.get('autoPush');```? Or then I don't think we need to bind it...

Reply:
yes, that's right. Thanks for catching this